src/WebJobs.Script/Utility.cs

```diff
@@ -879,6 +881,31 @@ public static bool IsResourceAzureBlobWithoutSas(Uri resourceUri)
             return true;
         }
 
+        public static string GetSasTokenExpirationDate(Uri resourceUri)
+        {
+            // Screen out URLs with an SAS token
+            var queryParams = HttpUtility.ParseQueryString(resourceUri.Query.ToLower());
+            if (queryParams != null && !string.IsNullOrEmpty(queryParams[SasTokenExpirationDate]))
+            {
+                return queryParams[SasTokenExpirationDate];
+            }
+
+            return string.Empty;
+        }
+
+        public static string GetSasTokenExpirationDateFromSasSignature(string azureWebJobsStorage)
+        {
+            var splitStorage = azureWebJobsStorage.Split(';');
+            var sasToken = splitStorage[splitStorage.Length - 1];
+            var queryParams = HttpUtility.ParseQueryString(sasToken);
+            if (queryParams != null && !string.IsNullOrEmpty(queryParams[SasTokenExpirationDate]))
+            {
+                return queryParams[SasTokenExpirationDate];
+            }
+
+            return string.Empty;
```

Comment:
I think if the query string entry is not present, we should return null instead of `string.empty`.

Reply:
You may also consider parsing the value and returning a `DateTime` instance or `null` from this method since this method sounds like it returns a Date (instead of Date string).

Reply:
Returned null! I decided to keep the return type as strings because we have to parse the queryParams[sasTokenExpirationDate] as a string explicitly first anyway before converting to a DateTime.